### ISSUE TITLE
rbenv init broke with fish 2.6.0

### ISSUE THREAD

**Issue by x4121** [rbenv/rbenv#1004]
This seems to be an error in fish-shell (IMHO), but I leave this here as a hint/note to others running into this error. Respectively to track the upstream progress in fish (maybe it's not a bug, but a feature. Then this could be a starting point :wink:)

Fish 2.6.0 changed setenv and the call `setenv PATH '/home/adg/.rbenv/shims' $PATH` from `rbenv init` isn't working any more.

**Comment by tuliocasagrande** [comment]
Related with: #986 
It's been merged, but it wasn't released yet.

**Comment by gobijan** [comment]
yup.. got this too: 

`status --is-interactive; and source (rbenv init -|psub)` 
produces
`setenv: Too many arguments`

**Comment by marktran** [comment]
Yep, it's fixed on HEAD. You can patch it or `brew unlink rbenv; brew install --HEAD rbenv` if you're using Homebrew.

**Comment by x4121** [comment]
Other workaround would be to replace
```
status --is-interactive; and source (rbenv init -|psub)
```
with
```
status --is-interactive; and source (rbenv init - | sed 's/setenv/set -gx/' | psub)
```

**Comment by jeremy** [comment]
Fixed by #986
